packages/firestore/src/platform_node/grpc_connection.ts

```diff
@@ -142,7 +103,27 @@ export class GrpcConnection implements Connection {
     const stub = this.ensureActiveStub(token);
     const rpc = stub[rpcMethod];
     assert(rpc != null, 'Unknown RPC: ' + rpcName);
-    return rpc.bind(stub);
+
+    const metadata = new grpc.Metadata();
+    if (token) {
+      for (const header in token.authHeaders) {
+        if (token.authHeaders.hasOwnProperty(header)) {
+          metadata.set(header, token.authHeaders[header]);
+        }
+      }
+    }
+    metadata.set('x-goog-api-client', X_GOOG_API_CLIENT_VALUE);
+    // This header is used to improve routing and project isolation by the
+    // backend.
+    metadata.set(
+      'google-cloud-resource-prefix',
+      `projects/${this.databaseInfo.databaseId.projectId}/` +
+        `databases/${this.databaseInfo.databaseId.database}`
+    );
+    const f = rpc.bind(stub);
+    return function() {
+      return f(arguments, metadata);
+    };
```

Comment:
I'd be a bit surprised if this works.  I think `arguments` will be an array-like object containing the arguments, which probably isn't what the RPC function expects?  I think you can do something like:

```
return function(...args) {
  f(...args, metadata);
}
```

and you could get rid of `f` and just do:

```
return function(...args) {
  rpc.bind(stub, ...args, metadata);
}
```

I _think_ that's right anyway...  I could be slightly off.

Reply:
Nice, that did the trick! ((It was definitely broken the way I had it.))